lib/portlayer/storage/image_cache.go

```diff
@@ -171,6 +169,12 @@ func (c *NameLookupCache) CreateImageStore(op trace.Operation, storeName string)
 	// Create the root image
 	scratch, err := c.DataStore.WriteImage(op, &Image{Store: store}, Scratch.ID, nil, "", nil)
 	if err != nil {
+		// if we failed here, remove the image store
+		op.Infof("Removing failed image store %s", storeName)
+		if e := c.DataStore.DeleteImageStore(op, storeName); e != nil {
+			op.Errorf("image store cleanup failed: %s", e.Error())
```

Comment:
This is what I want to ask as well. If eventually we didn't attach and also failed to delete the imagestore, is it possible to know that imagestore is invalid in next time imagestore creation?